spec/integration/associations_spec.rb

```diff
@@ -286,6 +286,35 @@ class Publisher < ActiveFedora::Base
       end
     end
 
+    describe "when fetching an existing object" do
+      before do
+        class Dog < ActiveFedora::Base; end
+        class BigDog < Dog; end
+        class Cat < ActiveFedora::Base; end
+        @dog = Dog.create()
+        @bigDog = BigDog.create()
+      end
+      it "should detect class mismatch" do
+        expect {
+          Cat.find(@dog.id)
+        }.to raise_error(ActiveFedora::ActiveFedoraError)
+      end
+
+      it "should not accept parent class into a subclass" do
+        expect {
+          BigDog.find(@dog.id)
+        }.to raise_error(ActiveFedora::ActiveFedoraError)
+      end
+
+      it "should accept a subclass into a parent class" do
+        # We could prevent this altogether since loading a subclass into
+        # a parent class (a bigDog into a Dog) would result in lost of 
+        # data if the subclass had more properties than the parent class
+        # and the object is saved back. 
+        expect(Dog.find(@bigDog.id).id).to eq @bigDog.id 
```

Comment:
Do we need to compare identifiers here, or is it sufficient that `Dog.find` doesn't raise here (which brings symmetry with its related tests above)?